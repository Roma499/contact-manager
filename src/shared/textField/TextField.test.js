import React from 'react';
import FormHelperText from '@material-ui/core/FormHelperText';
import FormControl from '@material-ui/core/FormControl';
import renderer from 'react-test-renderer';
import { shallow, mount } from 'enzyme';
import TextField from './TextField';

describe('render field without errors', () => {
  const props = {
    input: {
      name: 'firstName',
      onBlur: jest.fn(),
      onChange: jest.fn(),
      onDragStart: jest.fn(),
      onDrop: jest.fn(),
      onFocus: jest.fn(),
      value: ''
    },
    meta: {
      touched: false,
      error: undefined
    },
    name: 'firstName',
    type: 'text'
  };
  it('renders without crashing', () => {
    shallow(<TextField {...props} />);
  });
  it('renders correctly', () => {
    const tree = renderer.create(<TextField {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('contact fild FormControl prop error equal false', () => {
    const TextInputComponent = shallow(<TextField {...props} />).find(FormControl);
    expect(TextInputComponent.props().error).toEqual(false);
  });
});
describe('delete contact', () => {
  const props = {
    input: {
      name: 'firstName',
      onBlur: jest.fn(),
      onChange: jest.fn(),
      onDragStart: jest.fn(),
      onDrop: jest.fn(),
      onFocus: jest.fn(),
      value: ''
    },
    meta: {
      touched: true,
      error: 'This field is required'
    },
    name: 'firstName',
    type: 'text'
  };
  it('check if field has error', () => {
    const TextFieldComponent = mount(<TextField {...props} />);
    expect(TextFieldComponent.find(FormHelperText)).toHaveLength(1);
  });
  it('contact fild FormControl prop error equal true', () => {
    const TextInputComponent = shallow(<TextField {...props} />).find(FormControl);
    expect(TextInputComponent.props().error).toEqual(true);
  });
});
