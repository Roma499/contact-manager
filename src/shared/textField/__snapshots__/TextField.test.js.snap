// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render field without errors renders correctly 1`] = `
<div
  aria-describedby="name-error-text"
  className="MuiFormControl-root-1 contact-form-field"
>
  <label
    className="MuiFormLabel-root-10 MuiInputLabel-root-5 MuiInputLabel-formControl-6 MuiInputLabel-animated-9"
    data-shrink={false}
    htmlFor="name-error"
  />
  <div
    className="MuiInput-root-17 MuiInput-formControl-18 MuiInput-underline-21"
    onDragStart={[MockFunction]}
    onDrop={[MockFunction]}
  >
    <input
      aria-invalid={false}
      className="MuiInput-input-25"
      disabled={false}
      name="firstName"
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      required={false}
      type="text"
      value=""
    />
  </div>
</div>
`;
